barretenberg/cpp/CMakePresets.json

```diff
@@ -18,7 +18,7 @@
         "CMAKE_EXPORT_COMPILE_COMMANDS": "ON"
       },
       "cacheVariables": {
```

Comment:
Needed to make ASSERT_DEBUG work. doesnt affect our other asserts

Reply:
He, seems like we keep switching back and forth on whether the default builds have or dont have asserts.

1) Does this affect releases? (what provers/wasm will be using) Do we want that?
2) Can we/should we have an `-assert` variant like before?

Reply:
This isn't a flip-flop at all from my POV is that's where I wanted us to be - where most checks are runtime enforced and just a few ones in a tight loop are enabled for debug builds only. It was a bug more so as ASSERT_DEBUG doesn't do anything otherwise. release-assert feels rather niche.